internal/core/data/router.go

```diff
@@ -72,3 +80,212 @@ func LoadRestRoutes() *mux.Router {
 
 	return r
 }
+
+/*
+Return number of events in Core Data
+/api/v1/event/count
+*/
+func eventCountHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	switch r.Method {
+	case http.MethodGet:
+		count, err := count()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			LoggingClient.Error(err.Error(), "")
+			return
+		}
+
+		// Return result
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write([]byte(strconv.Itoa(count)))
+		if err != nil {
+			LoggingClient.Error(err.Error(), "")
+		}
+	}
+}
+
+/*
+Return number of events for a given device in Core Data
+deviceID - ID of the device to get count for
+/api/v1/event/count/{deviceId}
+*/
+func eventCountByDeviceIdHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	vars := mux.Vars(r)
+	id, err := url.QueryUnescape(vars["deviceId"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Problem unescaping URL: " + err.Error())
+		return
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		// Check device
+		count, err := countByDevice(id)
+		if err != nil {
+			LoggingClient.Error(fmt.Sprintf("error checking device %s %v", id, err))
+			switch err := err.(type) {
+			case types.ErrNotFound:
+				http.Error(w, err.Error(), http.StatusNotFound)
+				return
+			default: //return an error on everything else.
+				http.Error(w, err.Error(), http.StatusServiceUnavailable)
+				return
+			}
+		}
+
+		// Return result
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(strconv.Itoa(count)))
+		break
+	}
+}
+
+// Remove all the old events and associated readings (by age)
+// event/removeold/age/{age}
+func eventByAgeHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	vars := mux.Vars(r)
+	age, err := strconv.ParseInt(vars["age"], 10, 64)
+
+	// Problem converting age
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Error converting the age to an integer")
+		return
+	}
+
+	switch r.Method {
+	case http.MethodDelete:
+		LoggingClient.Info("Deleting events by age: " + vars["age"])
+
+		count, err := deleteByAge(age)
+		if err != nil {
+			LoggingClient.Error(err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		// Return the count
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(strconv.Itoa(count)))
```

Comment:
Is this valid JSON you are returning?

Reply:
No, it's just a number. I didn't write this. There's been a discussion between me, @SteveOss and the testing team about making all responses valid JSON types but that's a later effort.